core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -525,6 +525,22 @@ public static String resolveHostName(String hostname) throws UnknownHostExceptio
     return InetAddress.getByName(hostname).getCanonicalHostName();
   }
 
+  /**
+   * Resolves a given hostname IP address.
+   *
+   * @param hostname the input hostname, which could be an alias
+   * @return the hostname IP address, or null if it is null or empty
+   * @throws UnknownHostException if the given hostname cannot be resolved
+   */
+  @Nullable
+  public static String resolveIpAddress(String hostname) throws UnknownHostException {
+    if (hostname == null || hostname.isEmpty()) {
```

Comment:
Really good suggestions, I will make this change right away.

Reply:
@aaudiber after testing the method InetAddress.getByName with an empty string noticed that it resolve to 127.0.0.1, instead of throwing an UnknownHostException which is really weird...

Reply:
that's unfortunate, `Preconditions.checkArgument(!hostname.isEmpty())` sounds good to me 👍